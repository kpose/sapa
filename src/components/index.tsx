import LargeButton from './LargeButton/LargeButton';
import Started from './Started/Started';
import StartedCurrency from './StartedCurrency/StartedCurrency';
import CurrencyPicker from './CurrencyPicker/CurrencyPicker';
import FirstName from './FirstName/FirstName';
import Login from './Login/Login';
import Email from './Email/Email';
import HomeHeader from './HomeHeader/HomeHeader';
import HomeWallet from './HomeWallet/HomeWallet';
import WalletCard from './WalletCard/WalletCard';
import WavyHeader from './WavyHeader/WavyHeader';
import TabHeader from './TabHeader/TabHeader';
import Calender from './Calender/Calender';
import WalletInfo from './WalletInfo/WalletInfo';
import AddToWalletHeader from './AddToWalletHeader/AddToWalletHeader';
import AddToWalletBody from './AddToWalletBody/AddToWalletBody';
import SettingsItem from './SettingsItem/SettingsItem';
import SettingsBottom from './SettingsBottom/SettingsBottom';
import Spinner from './Spinner/Spinner';
import NetworkError from './NetworkError/NetworkError';
import ExpenseCard from './ExpenseCard/ExpenseCard';
import LoadingAnime from './Lottie/LoadingAnime';
import EmptyAnime from './Lottie/EmptyAnime';
import TransactionCard from './TransactionCard/TransactionCard';

export {
  LargeButton,
  Started,
  StartedCurrency,
  CurrencyPicker,
  FirstName,
  Login,
  Email,
  HomeHeader,
  HomeWallet,
  WalletCard,
  WavyHeader,
  TabHeader,
  Calender,
  WalletInfo,
  AddToWalletHeader,
  AddToWalletBody,
  SettingsItem,
  SettingsBottom,
  Spinner,
  NetworkError,
  ExpenseCard,
  LoadingAnime,
  EmptyAnime,
  TransactionCard,
};
