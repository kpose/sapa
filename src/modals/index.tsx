import CurrencyPicker from './CurrencyPicker/CurrencyPicker';
import AddWalletType from './AddWalletType/AddWalletType';
import AddWallet from './AddWallet/AddWallet';
import CameraModal from './CameraModal/CameraModal';
import CalenderPopup from './CalenderPopup/CalenderPopup';
import DatePickerModal from './DatePickerModal/DatePickerModal';
import NetworkErorModal from './NetworkErrorModal/NetworkErorModal';

export {
  CurrencyPicker,
  AddWalletType,
  AddWallet,
  CameraModal,
  CalenderPopup,
  DatePickerModal,
  NetworkErorModal,
};
