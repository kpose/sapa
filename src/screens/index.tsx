import Welcome from './Welcome/Welcome';
import Home from './Home/Home';

export {Welcome, Home};
