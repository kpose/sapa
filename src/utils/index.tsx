import * as sizes from './fonts';
import * as colors from './colors';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from './responsive';
import {last30Days, last7Days, walletTotal} from './totalCalculations';
import {modifiedExpenseArray} from './modifiedArrays';
import {modifiedIncomeArray} from './modifiedArrays';

export {
  sizes,
  wp,
  hp,
  colors,
  last30Days,
  last7Days,
  walletTotal,
  modifiedExpenseArray,
  modifiedIncomeArray,
};
