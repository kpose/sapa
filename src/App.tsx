import React, {useContext, useState, useEffect} from 'react';
import {store} from './redux/store';
import Routes from './navigation/routes';
import {Provider as ReduxProvider} from 'react-redux';
import {NavigationContainer} from '@react-navigation/native';
import {Provider as PaperProvider} from 'react-native-paper';
import {CombinedDarkTheme, CombinedLightTheme} from './utils/Theme';
import {ThemeContext} from './context/ThemeCotext';
import {createStackNavigator} from '@react-navigation/stack';
import {RouteStackParams} from '../src/definitions/navigationTypes';
import HomeStack from '../src/navigation/HomeStack';
import auth from '@react-native-firebase/auth';

const Stack = createStackNavigator<RouteStackParams>();

const App = () => {
  const [initializing, setInitializing] = useState(true);
  const [user, setUser] = useState();
  const [dark, setDark] = useState(true);
  const theme = dark ? CombinedDarkTheme : CombinedLightTheme;
  const toggleTheme = () => {
    setDark(!dark);
  };

  function onAuthStateChanged(user: any) {
    setUser(user);
    if (initializing) setInitializing(false);
  }

  useEffect(() => {
    const subscriber = auth().onAuthStateChanged(onAuthStateChanged);
    return subscriber;
  }, []);

  if (initializing) return null;

  return (
    <ThemeContext.Provider value={{theme, toggleTheme}}>
      <PaperProvider theme={theme}>
        <ReduxProvider store={store}>
          <NavigationContainer theme={theme}>
            {user ? (
              <Stack.Navigator
                screenOptions={{headerShown: false, gestureEnabled: false}}>
                <Stack.Screen name="Home" component={HomeStack} />
              </Stack.Navigator>
            ) : (
              <Routes />
            )}
          </NavigationContainer>
        </ReduxProvider>
      </PaperProvider>
    </ThemeContext.Provider>
  );
};

export default App;
