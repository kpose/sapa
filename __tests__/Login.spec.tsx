import * as React from 'react';
import renderer from 'react-test-renderer';
import {fireEvent, render, waitFor} from '@testing-library/react-native';
import {Login} from '../src/components';

jest.mock('@react-navigation/native');

describe('Testing login component', () => {
  const {} = render(<Login />);

  it('renders component on screen', () => {
    const tree = renderer.create(<Login />);
    expect(tree).toMatchSnapshot();
  });

  it('Should show password required', async () => {
    const email = 'user@mail.com';
    const {getByTestId} = render(
      <Login onBackPress={() => null} onButtonPress={() => null} />,
    );

    const emailInput = getByTestId('emailID');
    const button = getByTestId('buttonID');

    await waitFor(() => {
      fireEvent.changeText(emailInput, email);
      expect(emailInput.props.value).toBe(email);

      fireEvent.press(button);
      expect(getByTestId('passwordErrorID')).toBeDefined();
    });
  });

  it('Should call handle submit with email and password', async () => {
    const email = 'user@mail.com';
    const password = 'password1234';

    const expectedOutput = {
      email,
      password,
    };
    let output = {};

    const onSubmit = jest.fn(data => (output = data));

    const {getByTestId} = render(<Login onButtonPress={onSubmit} />);

    const emailInput = getByTestId('emailID');
    const button = getByTestId('buttonID');
    const passwordInput = getByTestId('passwordID');

    await waitFor(() => {
      fireEvent.changeText(emailInput, email);
      expect(emailInput.props.value).toBe(email);

      fireEvent.changeText(passwordInput, password);
      expect(emailInput.props.value).toBe(email);

      fireEvent.press(button);
      //expect(onSubmit).toHaveBeenCalledTimes(1);
      //expect(output).toEqual(expectedOutput);
    });
  });
});
