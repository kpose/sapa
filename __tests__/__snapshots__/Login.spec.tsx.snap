// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing login component renders component on screen 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "paddingBottom": 0,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "marginLeft": 75,
          "marginTop": 520.5,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(231, 231, 231)",
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
          },
          Object {
            "backgroundColor": "transparent",
            "marginTop": 26.5,
            "width": 637.5,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.26)",
            "bottom": 0,
            "height": 2,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "transform": Array [
              Object {
                "scaleY": 0.5,
              },
            ],
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "paddingBottom": 0,
              "paddingTop": 0,
            },
            Object {
              "minHeight": 64,
            },
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <Text
            numberOfLines={1}
            onLayout={[Function]}
            style={
              Object {
                "color": "#6200ee",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingLeft": 12,
                "paddingRight": 12,
                "position": "absolute",
                "textAlign": "left",
                "top": 34,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Email
          </Text>
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingLeft": 12,
                "paddingRight": 12,
                "position": "absolute",
                "textAlign": "left",
                "top": 34,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Email
          </Text>
        </View>
        <TextInput
          allowFontScaling={true}
          clearButtonMode="while-editing"
          editable={true}
          keyboardType="email-address"
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder=""
          placeholderTextColor="rgba(0, 0, 0, 0.54)"
          rejectResponderTermination={true}
          selectionColor="#6200ee"
          style={
            Array [
              Object {
                "flexGrow": 1,
                "margin": 0,
                "zIndex": 1,
              },
              Object {
                "paddingLeft": 12,
                "paddingRight": 12,
              },
              Object {
                "height": 64,
              },
              Object {
                "paddingBottom": 4,
                "paddingTop": 24,
              },
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              Array [
                Object {},
              ],
            ]
          }
          testID="emailID"
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(231, 231, 231)",
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
          },
          Object {
            "backgroundColor": "transparent",
            "marginTop": 26.5,
            "width": 637.5,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.26)",
            "bottom": 0,
            "height": 2,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "transform": Array [
              Object {
                "scaleY": 0.5,
              },
            ],
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "paddingBottom": 0,
              "paddingTop": 0,
            },
            Object {
              "minHeight": 64,
            },
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <Text
            numberOfLines={1}
            onLayout={[Function]}
            style={
              Object {
                "color": "#6200ee",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingLeft": 12,
                "paddingRight": 12,
                "position": "absolute",
                "textAlign": "left",
                "top": 34,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Password
          </Text>
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingLeft": 12,
                "paddingRight": 12,
                "position": "absolute",
                "textAlign": "left",
                "top": 34,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Password
          </Text>
        </View>
        <TextInput
          allowFontScaling={true}
          clearButtonMode="while-editing"
          editable={true}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onSubmitEditing={[Function]}
          placeholder=""
          placeholderTextColor="rgba(0, 0, 0, 0.54)"
          rejectResponderTermination={true}
          secureTextEntry={true}
          selectionColor="#6200ee"
          style={
            Array [
              Object {
                "flexGrow": 1,
                "margin": 0,
                "zIndex": 1,
              },
              Object {
                "paddingLeft": 12,
                "paddingRight": 12,
              },
              Object {
                "height": 64,
              },
              Object {
                "paddingBottom": 4,
                "paddingTop": 24,
              },
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              Array [
                Object {},
              ],
            ]
          }
          testID="passwordID"
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "marginLeft": -75,
          "marginTop": 240,
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#6200ee",
            "borderColor": "transparent",
            "borderRadius": 40,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 2,
            "height": 80,
            "justifyContent": "center",
            "minWidth": 64,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0.75,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 1.5,
            "width": 600,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 40,
              },
            ]
          }
          testID="buttonID"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    false,
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Array [
                      Object {
                        "fontSize": 17,
                        "fontWeight": "bold",
                      },
                    ],
                  ],
                ]
              }
            >
              Log In
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 20,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "fontSize": 18,
              "letterSpacing": -0.08,
            },
            Object {
              "marginRight": 15,
            },
          ],
        ]
      }
    >
      Get Started
    </Text>
  </View>
</View>
`;
